src/cljs/org/broadinstitute/firecloud_ui/page/workspace/workspace_data.cljs

```diff
@@ -42,15 +42,18 @@
                                          (make-button "All" :all)]}]]
         (if (zero? (count filtered-entities))
           (style/create-message-well "No entities to display.")
-          [table/Table
-           {:columns [{:header "Entity Type" :starting-width 100}
-                      {:header "Entity Name" :starting-width 100}
-                      {:header "Attributes" :starting-width 400}]
-            :data (map (fn [m]
-                         [(m "entityType")
-                          (m "name")
-                          (m "attributes")])
-                    filtered-entities)}])]))})
+          (let [attribute-keys (apply clojure.set/union (map (fn [e] (set (keys (e "attributes")))) filtered-entities))]
+            [table/Table
+             {:columns (concat
+                         [{:header "Entity Type" :starting-width 100 :sort-by :value}
+                          {:header "Entity Name" :starting-width 120 :sort-by :value}]
+                         (map (fn [ak] {:header ak :starting-width 100 :sort-by :value}) attribute-keys))
```

Comment:
FYI: Clojure style prefers the variable name `k` for a variable that will hold a key.